### PR TITLE
[express-serve-static-core] fix stack depth error on long path parameters

### PR DESCRIPTION
**Pull request by DetachHead**
Please fill in this template.

fix for this issue caused by my previous PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53513#issuecomment-870550063

i'm not really sure if this is the best way to detect the stack depth error. open to suggestions

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53513#issuecomment-870550063
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/express-serve-static-core/index.d.ts [hunk 1]
```diff
@@ -98,10 +98,12 @@ export type RequestHandlerParams<
     | ErrorRequestHandler<P, ResBody, ReqBody, ReqQuery, Locals>
     | Array<RequestHandler<P> | ErrorRequestHandler<P>>;
 
-type GetRouteParameter<RouteAfterColon extends string> = RouteAfterColon extends `${infer Char}${infer Rest}`
+type GetRouteParameter<RouteAfterColon extends string, Depth extends never[] = []> = Depth['length'] extends 10
```
Comment (ahejlsberg) [inline types/express-serve-static-core/index.d.ts:101]: This came up in a discussion on https://github.com/microsoft/TypeScript/issues/34933. I think a better way to write this utility type is:

```ts
type RemoveTail<S extends string, Tail extends string> = S extends `${infer P}${Tail}` ? P : S;
type GetRouteParameter<S extends string> = RemoveTail<RemoveTail<RemoveTail<S, `/${string}`>, `-${string}`>, `.${string}`>;
```

Comment (DetachHead) [reply types/express-serve-static-core/index.d.ts]: that's much better, thanks!

**Comment by daviduzumeri** [comment]
Any word on this? This is a super-blocker for updating packages right now.

**Review by ahejlsberg — APPROVED** [review]

**Comment by DetachHead** [comment]
Ready to merge

**Pull request merged**